secboot/secboot_tpm.go

```diff
@@ -308,6 +331,15 @@ func unlockEncryptedPartitionWithSealedKey(tpm *sb.TPMConnection, name, device,
 	return nil
 }
 
+// unlockEncryptedPartitionWithKey unlocks encrypted partition with the provided
+// key.
+func unlockEncryptedPartitionWithKey(name, device string, key []byte) error {
+
+	// TODO:UC20: call secboot.ActivateVolumeWithKey(name, device, key, &options)
+
+	return errors.New("not implemented")
```

Comment:
nil or not, this requires #9549 and subsequent bump to include the relevant API changes in secboot

Reply:
I think that returning non-nil here is confusing since it means, combined with the requirement that all encrypted devices have ubuntu-save, that no device with this specific branch of snapd can actually boot with encryption
If we returned nil at least we could boot until we implemented this in a follow-up PR after the secboot changes have landed

Reply:
Sorry, yes you're right. Locally I'm working with a duct taped branch that has all the required bits for now. The PR is blocked either way. The next step after this is called is an attempt to mount the device, which will fail. Maybe I could return some special error and a TODO so that we can try and land it without being blocked. Anyways, further changes in devicestate that expect /run/mnt/ubuntu-save mount would still require secboot.